tensorflow/lite/micro/python/py_cc_headers.bzl

```diff
@@ -0,0 +1,59 @@
+"Repository rule for creating a repository with the C headers of a python package."
+
+load("@tflm_pip_deps//:requirements.bzl", py_bzl_package = "requirement")
+
+_destdir = "include"
+
+_build = """
+cc_library(
+    name = "cc_library",
+    hdrs = glob(["%s/**/*.h"], allow_empty=False),
+    includes = ["%s"],
+    visibility = ["@//tensorflow/lite/micro:__subpackages__"],
+)
+""" % (_destdir, _destdir)
+
+def _make_include_path(bzl_package_path, py_package, prefix):
+    p = bzl_package_path.get_child(py_package)
+    if prefix:
+        for s in prefix.split("/"):
+            p = p.get_child(s)
+    return p
+
+def _tflm_py_cc_headers_impl(ctx):
+    # Symlink from the directory containing the headers, in the repository
+    # created by @python_rules pip_install(), to _destdir within this new
+    # repository.
+    any_file = py_bzl_package(ctx.attr.py_package) + ":BUILD.bazel"
+    bzl_package_path = ctx.path(Label(any_file)).dirname
+    src = _make_include_path(bzl_package_path, ctx.attr.py_package, ctx.attr.prefix)
```

Comment:
give an example in the comments about what this line is doing?

Reply:
Implementation refactored to be more clear, and comments added.